src/ansiblelint/yaml_utils.py

```diff
@@ -939,7 +945,11 @@ def version(self, value: str | tuple[int, int] | None) -> None:
         So, if a file does not include the directive, it sets this to None.
         But, None effectively resets the parsing version to YAML 1.2 (ruamel's default).
         """
-        self._yaml_version = value if value is not None else self._yaml_version_default
+        if value is not None:
+            self._yaml_version = value
+        elif hasattr(self, "_yaml_version_default"):
+            self._yaml_version = self._yaml_version_default
+        # We do nothing if the object did not have a previous default version defined
```

Comment:
I like the ternary expression that was here before. But this would work too:

```suggestion
        if value is not None:
            self._yaml_version = value
        else:
            self._yaml_version = self._yaml_version_default
```

Reply:
So, this would be my preference for this block:
```suggestion
        self._yaml_version = value if value is not None else self._yaml_version_default
```